src/v2/Components/FlashBanner/__tests__/FlashBanner.jest.tsx

```diff
@@ -1,32 +1,246 @@
+        data: { me: { id: "woot", canRequestEmailConfirmation: true } },
+      })
+
+      expect(wrapper.text()).toContain("Please verify your email address")
+    })
+
+    it("user seeing banner can click to trigger email confirmation message", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: { me: { id: "woot", canRequestEmailConfirmation: true } },
+        mutationResults: {
+          sendConfirmationEmail: {
+            confirmationOrError: {
+              unconfirmedEmail: "ceo@blackwater.biz",
+            },
+          },
+        },
+      })
+
+      expect(wrapper.text()).toContain("Please verify your email address")
+
+      wrapper
+        .find("button")
+        .first()
+        .prop("onClick")({} as any)
+      await flushPromiseQueue()
+      wrapper.update()
+
+      expect(wrapper.text()).toContain(
+        "An email has been sent to ceo@blackwater.biz"
+      )
+    })
+
+    it("user click on confirm email button is tracked", async () => {
+      const { wrapper, dispatch } = await getRelayWrapper({
+        data: { me: { id: "woot", canRequestEmailConfirmation: true } },
+        mutationResults: {
+          sendConfirmationEmail: {
+            confirmationOrError: {
+              unconfirmedEmail: "ceo@blackwater.biz",
+            },
+          },
+        },
+      })
+
+      expect(wrapper.text()).toContain("Please verify your email address")
+
+      wrapper
+        .find("button")
+        .first()
+        .prop("onClick")({} as any)
+
+      expect(dispatch).toHaveBeenCalledWith({
+        action_type: "Click",
+        subject: "Email Confirmation CTA",
+      })
+    })
+
+    it("user sees an error message if sendConfirmationEmail mutation fails", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: { me: { id: "woot", canRequestEmailConfirmation: true } },
+        mutationResults: {
+          sendConfirmationEmail: {
+            confirmationOrError: {
+              mutationError: {
+                error: "BadError",
+                message: "Something Bad",
+              },
+            },
+          },
+        },
+      })
+
+      expect(wrapper.text()).toContain("Please verify your email address")
+
+      wrapper
+        .find("button")
+        .first()
+        .prop("onClick")({} as any)
+      await flushPromiseQueue()
+      wrapper.update()
+
+      expect(wrapper.text()).toContain("Something went wrong")
+    })
+
+    it("a flash_message indicating an error supercedes the confirmation prompt", async () => {
+      const { wrapper } = await getRelayWrapper({
+        data: {
+          me: { id: "woot", canRequestEmailConfirmation: true },
+        },
+        queryString: "?flash_message=invalid_token",
+      })
+
+      expect(wrapper.text()).toContain("An error has occurred.")
+    })
+
+    it("user sees an error message to re-trigger verification if flash_message=expired_token", async () => {
```

Comment:
my previous comment got lost in the shuffle: https://github.com/artsy/force/pull/5667#discussion_r433268064

I still recommend removing this test